include/aws/common/byte_buf.h

```diff
@@ -233,36 +301,73 @@ AWS_COMMON_API
 int aws_byte_buf_cat(struct aws_byte_buf *dest, size_t number_of_args, ...);
 
 /**
- * Compares two aws_byte_cursor structures
- * Returns true if a has the same length as b and their buffers have the same bytes
- * (or both buffers are null). When both a and b are null the function returns true
+ * Compare two aws_byte_cursor structures.
+ * Return whether their contents are equivalent.
+ * NULL inputs are considered equal if both are NULL, or both lengths are 0.
```

Comment:
"NULL or empty (length = 0) inputs are considered equal if both are null or empty" perhaps? As it reads right now, it implies that null inputs have lengths.

Reply:
NULL is no longer accepted when comparing aws_byte_cursor, aws_byte_buf, aws_string, c_str.

It **is** still accepted in the underlaying aws_array_eq functions if the length is zero and I commented that like:

"NULL may be passed as the array pointer if its length is declared to be 0."